packages/builders/src/Assertions.ts

```diff
@@ -0,0 +1,22 @@
+import { Locale } from 'discord-api-types/v10';
+import { z } from 'zod';
+
+export const customIdPredicate = z.string().min(1).max(100);
+
+export const memberPermissionsPredicate = z
+	.string()
+	.refine((str) => !str.includes('.'), { message: 'Permissions must not contain decimal points' });
```

Comment:
Can't this have a union between number (int, turned to string), bigint (turned to string) and string (valid bigint)?

Reply:
Well, we're validating on the internal data field of the class, which will always store a string. If you have a better validation idea with that in mind, be my guest.

I guess `z.coerce.bigint().transform(num => num.toString())` could work and the error on the failed bigint coerce will be natural, since it just tries `BigInt(value)`